### PR TITLE
resistor-color: New exercise

### LINKED ISSUE #970: Add Resistor Color (#1) to replace Flatten Array as a side exercise
Currently, we have only 2 exercises that are unlocked by Hello World: Gigasecond and Flatten Array. 
Flatten Array [is problematic](https://github.com/exercism/ruby/issues/966), because it's basically a built-in Ruby method. And to implement it yourself, it's not a nice warmer-upper. 
There's currently no other candidate for this position on the track. 
Although Resistor Color is basically 'tutorial level', and therefore too easy for the track, it's a better warmer-upper than Flatten Array. So let's add it as a side exercise.

### PR DESCRIPTION
**Pull request by pgaspar**
Closes #970. Have a look at the canonical data [here](https://github.com/exercism/problem-specifications/tree/master/exercises/resistor-color). 🕺

### CODE REVIEW

**Comment by F3PiX** [comment]
Yay!!! Thanks Pedro :-)
One question:
I don't know nottin' about the generator. What are the consequences / problems of using `ResistorColor::COLORS` in the tests to return the array of colors?

I'm asking, because from a track anatomy point of view, and to make this suitable to be unlocked by HelloWorld, this should be a single method with a one-liner. And, it feels a bit weird that we have one method that's only there to satisfy the tests. 

Is that possible? Or would it mean that we have to manually regen the tests everytime something changes? 

Note: I'm working on a new description for Resistor Color (the series). I'd vote for leaving the description as is for now. Is that okay by everyone?

**Comment by kotp** [comment]
> Note: I'm working on a new description for Resistor Color (the series). I'd vote for leaving the description as is for now. Is that okay by everyone?

Thanks for mentioning it, I almost mentioned it in the approval.  Wasn't quite sure how to express it.

**Comment by pgaspar** [comment]
@F3PiX:

> I don't know nottin' about the generator. What are the consequences / problems of using ResistorColor::COLORS in the tests to return the array of colors?

Hmm, I think I can make that work - I think I understand what you want here.

> I'm asking, because from a track anatomy point of view, and to make this suitable to be unlocked by HelloWorld, this should be a single method with a one-liner. And, it feels a bit weird that we have one method that's only there to satisfy the tests.

Heh, this is the way to interpret the canonical specification, which specifies a second method to return the colors array ([here](https://github.com/exercism/problem-specifications/blob/master/exercises/resistor-color/canonical-data.json#L34-L39)). But I understand your points above and will change the generator to make that work :+1:

**Comment by F3PiX** [comment]
Briliant! Thanks. 

> Heh, this is the way to interpret the canonical specification, which specifies a second method to return the colors array

Yeah, it's not a bug, it's a feature :-) I'm still happy with asking to return the array of colors, and this is exactly how I envisioned it. 🙏 Supergreat.

File: exercises/resistor-color/.meta/generator/resistor_color_case.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def color_code_workload
+    assert_equal(expected, "ResistorColor.color_code(#{color.inspect})")
+  end
+
+  # Instead of specifying a `colors` method like the canonical data suggests
+  # we expect the student to define a COLORS constant.
+  def colors_workload
+    [
+      "expected = #{expected}",
+      "assert_equal expected, ResistorColor::COLORS"
```
Comment (F3PiX) [inline exercises/resistor-color/.meta/generator/resistor_color_case.rb]: 😍

**Comment by kotp** [comment]
Thank you @pgaspar ...

**Pull request merged**